ChopChop/Storage/Cache.swift

```diff
@@ -0,0 +1,71 @@
+import Foundation
+
+// https://www.swiftbysundell.com/articles/caching-in-swift/
+final class Cache<Key: Hashable, Value: CachableEntity> {
+    private let wrapped = NSCache<WrappedKey, Entry>()
+
+    func insert(_ value: Value, forKey key: Key) {
+        let entry = Entry(value: value)
+        wrapped.setObject(entry, forKey: WrappedKey(key))
+    }
+
+    func value(forKey key: Key) -> Value? {
+        let entry = wrapped.object(forKey: WrappedKey(key))
+        return entry?.value
+    }
+
+    func removeValue(forKey key: Key) {
+        wrapped.removeObject(forKey: WrappedKey(key))
+    }
+
+    subscript(key: Key) -> Value? {
+        get { value(forKey: key) }
+        set {
+            guard let value = newValue else {
+            // If nil was assigned using our subscript,
+            // then we remove any value for that key:
+                removeValue(forKey: key)
+                return
+            }
+
+            insert(value, forKey: key)
+        }
+    }
+
+    // NOTE: No type safety here because can compare any idss
```

Comment:
I was thinking along the lines of the fact that i can get a userInfoRecord, use its `id` and `updatedAt` to call the `getEntityIfCachedAndValid` method of the onlineRecipeCache, and get an OnlineRecipe. Which is abit strange. but not entirely sure whether this is considered an issue

But if the ids had the type safety stuff, this would not have been an issue